src/objects/Line.js

```diff
@@ -96,7 +101,8 @@ class Line extends Object3D {
 		_inverseMatrix.copy( matrixWorld ).invert();
 		_ray.copy( raycaster.ray ).applyMatrix4( _inverseMatrix );
 
-		const localThreshold = threshold / ( ( this.scale.x + this.scale.y + this.scale.z ) / 3 );
+		matrixWorld.decompose( _position, _quaternion, _scale );
+		const localThreshold = threshold / ( ( _scale.x + _scale.y + _scale.z ) / 3 );
```

Comment:
Using the scale in world space seems to be more correct indeed. Not sure why it was implemented differently in the first place.

Do you mind applying this patch to `Points.raycast()`, too?

Reply:
OK! I have applied his patch to Points.raycast().

Reply:
But `_scale.x` can be negative...

Why should the `threshold` for points and lines be scaled at all?

`threshold` is a world distance.

Reply:
That's a good point. I guess the scale factor should just be removed entirely?

Reply:
How about replacing _scale.x with Math.abs(_scale.x)  ?

Reply:
I think the primary question here is why should scale affect the global selection threshold. The threshold is there in the first place to make the otherwise infinitely thin lines selectable which does not seem like it should be impacted by object scale.

Reply:
The ray is converted to local space, the threshold also needs to be converted to local space to keep them the same scale.

_ray.copy( raycaster.ray ).applyMatrix4( _inverseMatrix )

Reply:
> the threshold also needs to be converted to local space to keep them the same scale.

This was also my thinking but considering the discussion it's probably best to not scale `threshold` at all.

Reply:
> Suppose point cloud A contains just one point at the origin and scale is 1.
> 
> Suppose point cloud B contains just one point at the origin and scale is 1000.
> 
> Why would you use a different `threshold` for detecting A vs B?

For this case, I don't think it need a different threshold. While, using a different scale doesn't do harm though. I'm thinking we could simply use the same scale, seems to be more strightforward. What do you think?